From fea692de44b4d8bf13f0cd07ad0835bacda3f210 Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Tue, 6 Mar 2018 13:56:55 +0200
Subject: [PATCH 3432/5242] MLK-17645: drm: imx: dcss: fix DTRC start issue

commit  c309ecfb6f2adea0005f9928defe192da01ffd6e from
https://source.codeaurora.org/external/imx/linux-imx.git

The following commit:

af01350 - MLK-17634-18: drm: imx: dcss: optimize context loading and DDR
bus load

introduced a regression. During my attempts to fix various green screen
issues, I modified the DTRC start routine by enabling the other register
bank, not the current one.

Unfortunately, this was committed by mistake...

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dcss/dcss-dtrc.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/imx/dcss/dcss-dtrc.c b/drivers/gpu/imx/dcss/dcss-dtrc.c
index a414fa7..4ae55ee 100644
--- a/drivers/gpu/imx/dcss/dcss-dtrc.c
+++ b/drivers/gpu/imx/dcss/dcss-dtrc.c
@@ -483,9 +483,9 @@ void dcss_dtrc_enable(struct dcss_soc *dcss, int ch_num, bool enable)
 		fdctl |= COMPRESSION_DIS;
 
 	dcss_dtrc_write(dtrc, ch_num, fdctl,
-			curr_frame * DTRC_F1_OFS + DCSS_DTRC_DCTL);
-	dcss_dtrc_write(dtrc, ch_num, fdctl | (enable ? CONFIG_READY : 0),
 			(curr_frame ^ 1) * DTRC_F1_OFS + DCSS_DTRC_DCTL);
+	dcss_dtrc_write(dtrc, ch_num, fdctl | (enable ? CONFIG_READY : 0),
+			curr_frame * DTRC_F1_OFS + DCSS_DTRC_DCTL);
 
 	ch->curr_frame = curr_frame;
 	ch->dctl = fdctl;
-- 
1.7.9.5

